js/ccf-app/src/global.ts

```diff
@@ -235,7 +235,15 @@ export interface EcdsaParams {
   hash: DigestAlgorithm;
 }
 
-export type SigningAlgorithm = RsaPkcsParams | EcdsaParams;
+/**
+ * EdDSA signature algorithm parameters.
+ */
+export interface EddsaParams {
+  name: "EdDSA";
+  hash: null;
```

Comment:
`hash: null` looks awkward, but it's necessary to keep the current design. If I remove it, codes like the following get a TS error.
```typescript
function foo(algorithm: SigningAlgorithm ) {
  console.log(algorithm.hash); // no property `hash` error
}
``` 

If we are happy to change the public interfaces including existing one, maybe we can do something like in future:
```typescript
export type AlgorithmName = "RSASSA-PKCS1-v1_5" | "ECDSA" | "EdDSA";

export interface SigningAlgorithm {
  name: AlgorithmName
  hash?: DigestAlgorithm
}
```

Reply:
We can change the public API in 3.0, since it's a major release.

Reply:
I made the change in [770bd07](https://github.com/microsoft/CCF/pull/4454/commits/770bd07f860ceb6e5b2c75fb15b8a3535de685fa).
Please check it just in case. Other than the change on `SigningAlgorithm` itself, the important change would be removing `RsaPkcsParams` and `EcdsaParams` from the public API.